numpy/core/src/umath/loops_unary_fp.dispatch.c.src

```diff
@@ -249,10 +456,179 @@ static void simd_@TYPE@_@kind@_@STYPE@_@DTYPE@
+            npyv_b@ssfx@ r2_@N@ = npyv_cvt_b@ssfx@_u@ssfx@(npyv_@kind@_@sfx@(v2_@N@));
+            npyv_b@ssfx@ r3_@N@ = npyv_cvt_b@ssfx@_u@ssfx@(npyv_@kind@_@sfx@(v3_@N@));
+            #if PACK_FACTOR == 8
+            npyv_b@ssfx@ r4_@N@ = npyv_cvt_b@ssfx@_u@ssfx@(npyv_@kind@_@sfx@(v4_@N@));
+            npyv_b@ssfx@ r5_@N@ = npyv_cvt_b@ssfx@_u@ssfx@(npyv_@kind@_@sfx@(v5_@N@));
+            npyv_b@ssfx@ r6_@N@ = npyv_cvt_b@ssfx@_u@ssfx@(npyv_@kind@_@sfx@(v6_@N@));
+            npyv_b@ssfx@ r7_@N@ = npyv_cvt_b@ssfx@_u@ssfx@(npyv_@kind@_@sfx@(v7_@N@));
+            #endif // PACK_FACTOR == 8
+        #endif // @PREPACK@ && (@ssfx@ == 32 || @ssfx@ == 64)
+        #endif // @unroll@ > @N@
+        /**end repeat3**/
+
+        /**begin repeat3
+         * #N  = 0, 1, 2, 3#
+         */
+        #if @unroll@ > @N@
+        #if @DTYPE@ == CONTIG
+            #if @PREPACK@ && (@ssfx@ == 32 || @ssfx@ == 64)
+                // Nothing to do, results already packed
+            #else
+                // Pack results
+                #if PACK_FACTOR == 4
+                npyv_u8 r_@N@ = npyv_cvt_u8_b8(npyv_pack_b8_b32(r0_@N@, r1_@N@, r2_@N@, r3_@N@));
+                #elif PACK_FACTOR == 8
+                npyv_u8 r_@N@ = npyv_cvt_u8_b8(npyv_pack_b8_b64(r0_@N@, r1_@N@, r2_@N@, r3_@N@,
+                                                                r4_@N@, r5_@N@, r6_@N@, r7_@N@));
+                #endif // PACK_FACTOR == 8
+            #endif // @PREPACK@ && (@ssfx@ == 32 || @ssfx@ == 64)
+
+            npyv_store_u8(op + vstep * PACK_FACTOR * @N@, r_@N@);
+
+        #else // @DTYPE@ == CONTIG
+            #if @PREPACK@ && (@ssfx@ == 32 || @ssfx@ == 64)
+                // Results are packed, so we can just loop over them
+                npy_uint8 lane_@N@[npyv_nlanes_u8];
+                npyv_store_u8(lane_@N@, r_@N@);
+                for (int ln=0; (ln * sizeof(npyv_lanetype_@sfx@)) < npyv_nlanes_u8; ++ln){
+                    op[(ln + @N@ * PACK_FACTOR * vstep) * ostride] = lane_@N@[ln * sizeof(npyv_lanetype_@sfx@)];
+                }
+            #else
+                // Results are not packed.  Use template to loop.
+                /**begin repeat4
+                 * #R = 0, 1, 2, 3, 4, 5, 6, 7#
+                 */
+                #if @R@ < PACK_FACTOR
+                npy_uint8 lane@R@_@N@[npyv_nlanes_u8];
+                npyv_store_u8(lane@R@_@N@, npyv_reinterpret_u8_u@ssfx@(r@R@_@N@));
+                op[(0 + (@R@ + @N@ * PACK_FACTOR) * vstep) * ostride] = lane@R@_@N@[0 * sizeof(npyv_lanetype_@sfx@)];
+                op[(1 + (@R@ + @N@ * PACK_FACTOR) * vstep) * ostride] = lane@R@_@N@[1 * sizeof(npyv_lanetype_@sfx@)];
+                #if npyv_nlanes_@sfx@ == 4
+                op[(2 + (@R@ + @N@ * PACK_FACTOR) * vstep) * ostride] = lane@R@_@N@[2 * sizeof(npyv_lanetype_@sfx@)];
+                op[(3 + (@R@ + @N@ * PACK_FACTOR) * vstep) * ostride] = lane@R@_@N@[3 * sizeof(npyv_lanetype_@sfx@)];
+                #endif
```

Comment:
I can understand not having hardware on CI to cover lines 518-582, but are we missing a test for 577-579?

Reply:
Looking into it